server/controllers/chartsController/index.js

```diff
@@ -122,35 +122,92 @@ const getAllSubjects = async ({ dataDb, chart, userAccess, filters }) => {
       )
       .toArray()
   } else {
-    const { mongoAggregateQueryForFilters, activeFilters } =
-      mongoQueryFromFilters(filters)
+    const {
+      mongoAggregateQueryForIncludedCriteria,
+      mongoAggregateQueryForFilters,
+      mongoQueryForSocioDemographics,
+      activeFilters,
+    } = mongoQueriesFromFilters(filters, userAccess)
+    const mapForResultsFromFilters = new Map()
     const filteredSubjects = []
-    const criteriaCursor = await dataDb.collection(collections.toc).find(
-      {
-        assessment: INCLUSION_EXCLUSION_CRITERIA_FORM,
-        study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
-      },
-      {
-        projection: ALL_SUBJECTS_MONGO_PROJECTION,
-      }
-    )
+    const criteriaCursor = await dataDb
+      .collection(collections.toc)
+      .aggregate(mongoAggregateQueryForFilters)
 
     for await (let doc of criteriaCursor) {
-      const data = await dataDb
-        .collection(doc.collection)
-        .aggregate(mongoAggregateQueryForFilters)
-      const subjectCriteriaData = await data.next()
-      const isSubjectInFilteredQuery = activeFilters
-        .map(
-          (requestedFilter) => subjectCriteriaData[requestedFilter].length > 0
-        )
-        .every(Boolean)
-
-      if (isSubjectInFilteredQuery) {
-        filteredSubjects.push(doc.subject)
```

Comment:
this code that is removed gets us the filtered subjects for the existing filters. If we can get a new collection of filtered subject for the new sex at birth filters we can compare these `filteredSubjects` with the `sexAtBirthFilteredSubjects` and leave everything else the same. I think?

Reply:
This code is no longer removed, just expanded with the additional queries